src/Symfony/Component/PasswordHasher/Hasher/UserPasswordHasher.php

```diff
@@ -29,6 +34,16 @@ public function __construct(PasswordHasherFactoryInterface $hasherFactory)
 
     public function hashPassword(UserInterface $user, string $plainPassword): string
```

Comment:
This class has not been released yet. Can we upgrade the signature to the desired type declarations already?

Reply:
Using this class requires to follow an upgrade path as well (developers will need to use `password_hashers` for using it). Accepting both the old type and the new one allow upgrading one step at a time without breakage